client/base_client.go

```diff
@@ -137,7 +138,7 @@ func (c *baseClient) leaderLoop() {
 		}
 
 		if err := c.updateLeader(); err != nil {
-			log.Error("[pd] failed updateLeader", zap.Error(err))
+			log.Error("[pd] failed updateLeader", zap.Error(errs.ErrGRPCSend.FastGenByArgs()), zap.NamedError("cause", err))
```

Comment:
Update leader error may not be grpc send error. What about creating a new error type to solve the function with too many possible error sources?

Reply:
You're right. Already working on it.